pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -2035,6 +2069,50 @@ func triggerPublishHwInfo(ctxPtr *zedagentContext) {
 	triggerPublishHwInfoToDest(ctxPtr, AllDest)
 }
 
+func handleClusterUpdateStatusCreate(ctxArg interface{}, key string,
+	configArg interface{}) {
+	handleClusterUpdateStatusImpl(ctxArg, key, configArg, nil)
+}
+func handleClusterUpdateStatusModify(ctxArg interface{}, key string,
+	configArg interface{}, oldStatusArg interface{}) {
+	handleClusterUpdateStatusImpl(ctxArg, key, configArg, oldStatusArg)
+}
+func handleClusterUpdateStatusImpl(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+	ctx, ok := ctxArg.(*zedagentContext)
+	if !ok {
+		log.Fatalf("handleClusterUpdateStatusImpl invalid type in ctxArg: %v", ctxArg)
```

Comment:
Do we really have to log fatal ? Remember fatal will reboot the device.  FYI, Rebooting the device during upgrade may not work out well. I will leave it to you.

Reply:
Also we cannot fatal just for invalid args